k2/csrc/cuda/debug.cuh

```diff
@@ -0,0 +1,169 @@
+// k2/csrc/cuda/errors.h
+
+// Copyright (c) 2020, Xiaomi Corporation ( authors: Meixu Song )
+
+// See LICENSE for clarification regarding multiple authors
+
+#ifndef K2_CSRC_CUDA_ERRORS_H_
+#define K2_CSRC_CUDA_ERRORS_H_
+
+#include <cstdio>
+
+#include "k2/csrc/cuda/arch.h"
+
+namespace k2 {
+
+// In debug mode and include this header,
+// define K2_MAKE_ERROR_CHECK to turn on error checking
+#if (!defined(NDEBUG) && !defined(K2_MAKE_ERROR_CHECK))
+  #define K2_MAKE_ERROR_CHECK
+#endif
+
+/**
+ * \brief A static assertion
+ * \param exp the compile-time boolean expression that must be true
+ * \param msg an error message if exp is false
```

Comment:
The comment format is not consistent with our conventions, we use `@param` in other files and no `\brief`

Reply:
The \brief is OK with me as it's likely more correct.  I am using a kind of retarded doxygen that I don't attempt to compile.

Reply:
Thx! I turn to learn about doxygen and will change to some suggesting conventions in the coming update.